### DISCUSSION TITLE
low priority modules to hide when bar has too much and overflows

### DISCUSSION THREAD

**Discussion by gavsiu** [polybar/polybar#2722] [Ideas]
I guess I have too much on my bar.

It would be nice if I could designate certain bars as low priority and have them not appear if the bar gets too long. For instance, if my window title is too long, I'd rather have it show than media player (low priority).

**Comment by patrick96** [comment]
Something like this has been an open feature request for a long time: #395

The request itself is for priorities on the block level (left, right, center), I think the effort for making this on a per module basis would be a bit bigger because the renderer doesn't know about modules.

But I think the bigger effort would be the hiding logic anyway. The core of it would be a "scheduling" algorithm that given a list of blocks (or modules) with priorities, figures out which blocks need to be cut off and by how much and where they would be placed on the bar. The renderer would then also need to be adapted to place each block at those positions and cut them off correctly.
